libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/MapboxRerouteController.kt

```diff
@@ -214,41 +221,59 @@ internal class MapboxRerouteController @VisibleForTesting constructor(
         callback: NavigationRerouteController.RoutesCallback,
         routeOptions: RouteOptions
     ) {
-        requestId = directionsSession.requestRoutes(
-            routeOptions,
-            object : NavigationRouterCallback {
-                override fun onRoutesReady(
-                    routes: List<NavigationRoute>,
-                    routerOrigin: RouterOrigin
-                ) {
-                    mainJobController.scope.launch {
-                        state = RerouteState.RouteFetched(routerOrigin)
-                        state = RerouteState.Idle
-                        callback.onNewRoutes(routes, routerOrigin)
-                    }
-                }
+        rerouteScope.launch {
+            val result = suspendCancellableCoroutine<Pair<List<NavigationRoute>, RouterOrigin>?> { cont ->
+                cont.invokeOnCancellation { onRequestInterrupted() }
+                requestId = directionsSession.requestRoutes(
+                    routeOptions,
+                    object : NavigationRouterCallback {
+                        override fun onRoutesReady(
+                            routes: List<NavigationRoute>,
+                            routerOrigin: RouterOrigin
+                        ) {
+                            mainJobController.scope.launch {
+                                state = RerouteState.RouteFetched(routerOrigin)
+                                state = RerouteState.Idle
+                            }
+                            if (cont.isActive) {
```

Comment:
`cont.isActive` are likely not necessary. If I am not mistaken, execution will not continue after `suspendCancellableCoroutine` if the scope was canceled.